docs(matrix): add system requirements doc DOC-350 (#547)
armory/docs: content/en/docs/installation/system-requirements.md

---
title: System Requirements for Armory Enterprise
linkTitle: System Requirements
description: >
  Hardware and other requirements to run Armory Enterprise
---

## Browsers

> You can use any modern browser to interact with the UI.

Reviewer: Do we want to put a limit on `modern` or is it generally understood what that means?

|------------|--------------------------|-----------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Redis      | All supported versions   | All services that require a backing store | The DB version refers to external Redis instances. Supported only for services that still require Redis for a feature, such as Gate sessions. Redis is not supported as a core persistent storage engine. Although Armory Enterprise can deploy internal Redis instances, do not use these for production deployments. Armory recommends only using them for testing and proof-of-concept deployments. |

> | MySQL      | MySQL 5.7 (or Aurora)    | Clouddriver, Front50, Orca                          |                                                                                                                                                                                                                                                                                                                                                                                                       |

Reviewer: ```suggestion
| MySQL      | MySQL 5.7 (or Aurora)    | Clouddriver, Front50, Orca, Fiat, Echo                |                                                                                                                                                                                                                                                                                                                                                                                                       |
```

> | PostgreSQL | PostgreSQL 10.0 or later | Clouddriver                                         |                                                                                                                                                                                                                                                                                                                                                                                                       |

Reviewer: ```suggestion
| PostgreSQL | PostgreSQL 10.0 or later | Clouddriver, Front50, Orca, Fiat                     |                                                                                                                                                                                                                                                                                                                                                                                                       |
```

Armory recommends using MySQL or PostgreSQL as the backing store when possible for production instances. For services where this is not an option, use an external Redis instance.

## Hardware requirements

## Java

Java 11 is required to run Halyard (one method of configuring and managing Armory Enterprise) on your machine.

## Kubernetes version

> Armory Enterprise is a collection of services that run in a Kubernetes cluster. This section defines the versions that you can run Armory Enterprise on, not where you can deploy your applications. For information about what versions of Kubernetes you can deploy to, see the [Product Compatibility Matrix]({{< ref "armory-platform-matrix.md" >}}).

Reviewer 2: ```suggestion
Armory Enterprise is a collection of services that run in a Kubernetes cluster. This section defines the versions that you can run Armory Enterprise on, not where you can deploy your applications. For information about what versions of Kubernetes you can deploy to, see the [Product Compatibility Matrix]({{< ref "armory-platform-matrix#container-as-a-service-platforms" >}}).
```
Since you specifically mention what versions of K8s you can deploy to, how about linking directly to the relevant section on the PCM page?

| MySQL    | MySQL 5.7 (or Aurora)  | All supported versions | Clouddriver, Front50, Orca                          |                                                                                                                            |
| PostgreSQL    | PostgreSQL 10.0 or later  | 2.24.0 or later | Clouddriver                          |                                                                                                                            |

Armory recommends using MySQL or PostgreSQL as the backing store when possible for production instances of Spinnaker. For other services, use an external Redis instance for production instances of Spinnaker.

## Hardware requirements

Armory recommends a minimum of 3 nodes that match the following profile:

* **CPUS**: 8
* **Memory (GiB)**: 32

## kubectl

> To install and manage Armory Enterprise, any actively maintained version of `kubectl` is supported unless you want to use the [Armory Kustomize Repo](https://github.com/armory/spinnaker-kustomize-patches). The Kustomize Repo is a collection of Kustomize files that help you configure Armory Enterprise. The repo supports the following versions of `kubectl`: 1.16 to 1.19.

Reviewer 3: This may need rewording as the Armory Kustomize Repo uses kubectl as well but it is highly suggested to use that repo

---
title: System Requirements for Armory Enterprise
linkTitle: System Requirements
weight: 1
description: >
  Armory Enterprise is a collection of services that run in a Kubernetes cluster. In addition to the cluster, other requirements, such as storage, need to be met to run Armory Enterprise for production environments.
---

The requirements described on this page are meant as a minimum starting point for installing Armory Enterprise. You may need to increase the resources based on the number of applications, pipelines, and executions. Work with your IT organization to make sure that the requirements are met.

## Installation targets

> This section defines where you can run Armory Enterprise, not where you can deploy your applications. For information about where you can deploy applications to, see the [Product Compatibility Matrix]({{< ref "armory-enterprise-matrix#deployment-targets" >}}).

Armory Enterprise can be installed on any [certified Kubernetes cluster](https://www.cncf.io/certification/software-conformance/) that meets the following version requirements:

* **Minimum version**: 1.16
* **Maximum version**: 1.20
<!-- track EKS versions -->

> You install Armory Enterprise using the Armory Operator (a Kubernetes operator), which has the following requirements:

Reviewer 2: ```suggestion
You install Armory Enterprise using the [Armory Operator]({{< ref "armory-operator" >}}) (a Kubernetes operator), which has the following requirements:
```

| -------- | ---------------------- | ---------------------- | --------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------- |
| Redis    | All supported versions | All supported versions | All Spinnaker services that require a backing store | The DB versions refer to external Redis instances. Supported only for services that still require Redis for a feature, such as Gate sessions. Redis is not supported as a core persistent storage engine. Although Spinnaker deploys internal Redis instances, do not use these instances for production deployments. Armory recommends only using them for testing and proof-of-concept deployments. |
| MySQL    | MySQL 5.7 (or Aurora)  | All supported versions | Clouddriver, Front50, Orca                          |                                                                                                                            |

> | PostgreSQL    | PostgreSQL 10.0 or later  | 2.24.0 or later | Clouddriver                          |                                                                                                                            |

Reviewer 2: MySQL and PostgreSQL version are now parameters in config.toml.
mysql-version="5.7; AWS Aurora"
postgresql-version="10+"

Author: Cool! Thanks for thinking ahead

> Armory recommends using MySQL or PostgreSQL as the backing store when possible for production instances of Spinnaker. For other services, use an external Redis instance for production instances of Spinnaker.

Reviewer 2: ```suggestion
Armory recommends using MySQL or PostgreSQL as the backing store when possible for production instances of Spinnaker. For other services, use an external Redis instance for production instances of Armory Enterprise.
```

## Hardware requirements

Armory recommends a minimum of 3 nodes that match the following profile:

* **CPUS**: 8
* **Memory (GiB)**: 32

## kubectl

> To install and manage Armory Enterprise, Armory recommends using the [Armory Kustomize Repo](https://github.com/armory/spinnaker-kustomize-patches) and tailoring the Kustomize files to meet the requirements of your instance and environment. The repo supports the following versions of `kubectl`: 1.16 to 1.19.

Reviewer 2: Rather than pointing to the repo, how about pointing to installation/armory-operator/op-config-kustomize.md ?

```suggestion
To install and manage Armory Enterprise, Armory recommends using the [Armory Operator with Kustomize]({{< ref "op-config-kustomize" >)}} and tailoring the Kustomize files to meet the requirements of your instance and environment. This installation method supports the following versions of `kubectl`: 1.16 to 1.19.
```

<!-- this is cause of the go-yaml bug: https://github.com/kubernetes-sigs/kustomize/issues/3605 -->

> It is possible to use the Operator to install Armory Enterprise without the Kustomize repo. In that case, any actively maintained version of `kubectl` is supported.

Reviewer 2: ```suggestion
It is possible to use the Operator to [install Armory Enterprise without the Kustomize repo]({{< ref "op-config-manifest" >}}). In that case, any actively maintained version of `kubectl` is supported.
```

---
title: System Requirements for Armory Enterprise
linkTitle: System Requirements
description: >
  Armory Enterprise is a collection of services that run in a Kubernetes cluster. In addition to the cluster, other requirements, such as storage, need to be met to run Armory Enterprise for production environments.
---

The requirements described on this page are meant as a minimum starting point for installing Armory Enterprise. You may need to increase the resources based on the number of pipelines and executions.

## Installation targets

> This section defines where you can run Armory Enterprise, not where you can deploy your applications. For information about where you can deploy applications to, see the [Product Compatibility Matrix]({{< ref "armory-enterprise-matrix#deployment-targets" >}}).

Armory Enterprise can be installed on any [certified Kubernetes cluster](https://www.cncf.io/certification/software-conformance/).

> * **Minimum version**: Some TBD version

Reviewer: 1.16 as of this writing (lower bound for EKS)

> * **Maximum version**: Some TBD version

Reviewer: 1.20 as of this writing (upper bound for EKS)

I think long-term we should follow what EKS supports until we see customers overwhelmingly on another solution

> To install Armory Enterprise, you can use either the Armory Operator (a Kubernetes operator) or Halyard. If you are installing Armory Enterprise for the first time, Armory recommends using the Armory Operator. Note that you must be able to apply Kubernetes manifests, either directly using `kubectl` commands from your machine or another method.

Reviewer: ```suggestion
To install Armory Enterprise, you can use either the Armory Operator (a Kubernetes operator). Note that you must be able to apply Kubernetes manifests, either directly using `kubectl` commands from your machine or another method.
```
